### ISSUE TITLE
Limiting package versions to profile request range

### ISSUE THREAD

**Issue by davidlatwe** [mottosso/allzpark#127]
### Problem

In "Packages" tab, user can change package version with a drop-down list. But the version listed in there is a full list of all installed versions, which can be easily go out side of profile's request range.

E.g. Maya package family has version 2016-2020 installed, but the profile however :

```python
requires = ["~maya==2020"]
```

And if you change the version into, say 2018, Maya package resolve would fail.

### Proposal

We should avoid exposing those "dead-ends" on interface. Only show versions that are not exceeding profile's requirement.

Also, how about disable those non-application package's version selecting ? I think this could save some compute resource, plus, changing non-application package's version is somewhat not practical.

**Comment by mottosso** [comment]
Oo, but I wonder whether we can know which are compatible, without running a full resolve. Since each resolve can take quite some time, it could delay showing that drop-down for many seconds, maybe minutes. What have you got in mind?

**Comment by davidlatwe** [comment]
> but I wonder whether we can know which are compatible, without running a full resolve.

You mean to be able to limit the version range without running full resolve for each combination ?

**Comment by davidlatwe** [comment]
If so, I am thinking to jsut pass the version range of profile's package request in `requires` when listing pacakges, like here

https://github.com/mottosso/allzpark/blob/a2e48e02a10124b3715d07130de58bfc589a2792/allzpark/model.py#L380

And if that package is not meant to be an application (not weak referenced), skip verison listing. :P

**Comment by davidlatwe** [comment]
Say we have this profile :
```
# profile package
requires = [
    "maya_usd",
    "~maya==2018|==2020",
]
```
And the Maya package, maybe like this :
```
# maya package
requires = [
    "foo",
]
```

Only Maya will have version `2018`, `2020` listed, versions of `maya_usd` and the dependency `foo` will be fixed.

**Comment by davidlatwe** [comment]
Got a working draft now (`PackagesModel` and `ApplicationModel` also been merged into one), once the bug described in https://github.com/mottosso/allzpark/pull/126#issuecomment-724846018 gets resovled, I could submit another PR which will supersede #126 .

**Comment by davidlatwe** [comment]
Resolved with #129.
